src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -157,18 +167,21 @@ private boolean matches(Offer offer, NMProfile profile) {
         if (mem < 0) {
             LOGGER.error("No mem resource present");
         }
+        if (ports < 0) {
+            LOGGER.error("No port resources present");
+        }
 
         Map<String, String> requestAttributes = new HashMap<>();
 
         if (taskUtils.getAggregateCpus(profile) <= cpus
                 && taskUtils.getAggregateMemory(profile) <= mem
                 && SchedulerUtils.isMatchSlaveAttributes(offer,
-                requestAttributes)) {
+                requestAttributes) && ports >= 3) {
```

Comment:
Said magic number appears in on other location as well.  There's a few other constants appearing in multiple files.  Is it worth adding a constants class to avoid changes to one and not another?